src/System.Memory/src/System/Span.cs

```diff
@@ -233,6 +233,109 @@ internal Span(Pinnable<T> pinnable, IntPtr byteOffset, int length)
         }
 
         /// <summary>
+        /// Clears the contents of this span.
+        /// </summary>
+        public unsafe void Clear()
+        {
+            var length = _length;
```

Comment:
You'll want to use the cached length in the "length == 0" check below.

Also, this isn't an allowed use of "var" - the actual type doesn't appear on the RHS.